crates/comm/src/peer/mod.rs

```diff
@@ -217,6 +243,7 @@ impl Peer {
         let tosend = ProtocolPacket {
             packet_type: Some(gossip),
         };
+        // TODO: why is there a clone here? does the function above need one too?
         self.send_packet(tosend.clone()).await?;
```

Comment:
If it seems redundant, it probably is. Remove it and see if anything breaks.
```suggestion
        self.send_packet(tosend).await?;
```